### ISSUE TITLE
AADRoleEligibilityScheduleRequest: {"errorCode":"PermissionScopeNotGranted","message":"Authorization failed due to missing permission scope RoleEligibilitySchedule.ReadWrite.Directory,RoleManagement.ReadWrite.Directory.","instanceAnnotations":[]} "Error during Export:"

### ISSUE THREAD

**Issue by skghq** [microsoft/Microsoft365DSC#3940]
### Description of the issue

I have already ran: `Update-M365DSCAllowedGraphScopes -All -Type Read`

Command:
`Export-M365DSCConfiguration -Components @("AADRoleEligibilityScheduleRequest") -Credential $Credential -Path $FilePath -FileName $FileName`

Stack Trace:
`
[2023/11/23 09:41:25]
{InvalidOperation}
System.Exception: [UnknownError] : {"errorCode":"PermissionScopeNotGranted","message":"Authorization failed due to missing permission scope RoleEligibilitySchedule.ReadWrite.Directory,RoleManagement.ReadWrite.Directory.","instanceAnnotations":[]}
"Error during Export:"
at Get-MgBetaRoleManagementDirectoryRoleEligibilityScheduleRequest<Process>, C:\Program Files\WindowsPowerShell\Modules\Microsoft.Graph.Beta.Identity.Governance\2.9.0\exports\ProxyCmdletDefinitions.ps1: line 113122
at Export-TargetResource, C:\Program Files\WindowsPowerShell\Modules\Microsoft365DSC\1.23.1115.1\DSCResources\MSFT_AADRoleEligibilityScheduleRequest\MSFT_AADRoleEligibilityScheduleRequest.psm1: line 743
at Start-M365DSCConfigurationExtract, C:\Program Files\WindowsPowerShell\Modules\Microsoft365DSC\1.23.1115.1\modules\M365DSCReverse.psm1: line 615
at Export-M365DSCConfiguration, C:\Program Files\WindowsPowerShell\Modules\Microsoft365DSC\1.23.1115.1\modules\M365DSCUtil.psm1: line 1320
at <ScriptBlock>, C:\Scripts\<<<redacted>>>\Office365_Backup_Debug_Permissions.ps1: line 59
<<<redacted>>>@<<<redacted>>>.onmicrosoft.com
TenantId: <<<redacted>>>.onmicrosoft.com
`

### Microsoft 365 DSC Version

1.23.1115.1

### Which workloads are affected

Azure Active Directory

### The DSC configuration

```powershell
Export-M365DSCConfiguration -Components @("AADRoleEligibilityScheduleRequest") -Credential $Credential -Path $FilePath -FileName $FileName
```

### Verbose logs showing the problem

```powershell
[2023/11/23 09:41:25]
{InvalidOperation}
System.Exception: [UnknownError] : {"errorCode":"PermissionScopeNotGranted","message":"Authorization failed due to missing permission scope RoleEligibilitySchedule.ReadWrite.Directory,RoleManagement.ReadWrite.Directory.","instanceAnnotations":[]}
"Error during Export:"
at Get-MgBetaRoleManagementDirectoryRoleEligibilityScheduleRequest<Process>, C:\Program Files\WindowsPowerShell\Modules\Microsoft.Graph.Beta.Identity.Governance\2.9.0\exports\ProxyCmdletDefinitions.ps1: line 113122
at Export-TargetResource, C:\Program Files\WindowsPowerShell\Modules\Microsoft365DSC\1.23.1115.1\DSCResources\MSFT_AADRoleEligibilityScheduleRequest\MSFT_AADRoleEligibilityScheduleRequest.psm1: line 743
at Start-M365DSCConfigurationExtract, C:\Program Files\WindowsPowerShell\Modules\Microsoft365DSC\1.23.1115.1\modules\M365DSCReverse.psm1: line 615
at Export-M365DSCConfiguration, C:\Program Files\WindowsPowerShell\Modules\Microsoft365DSC\1.23.1115.1\modules\M365DSCUtil.psm1: line 1320
at <ScriptBlock>, C:\Scripts\<<<redacted>>>\Office365_Backup_Debug_Permissions.ps1: line 59
<<<redacted>>>@<<<redacted>>>.onmicrosoft.com
TenantId: <<<redacted>>>.onmicrosoft.com
```

### Environment Information + PowerShell Version

```powershell
OsName               : Microsoft Windows Server 2019 Standard
OsOperatingSystemSKU : StandardServerEdition
OsArchitecture       : 64-bit
WindowsVersion       : 1809
WindowsBuildLabEx    : 17763.1.amd64fre.rs5_release.180914-1434
OsLanguage           : en-US
OsMuiLanguages       : {en-US}

Key   : PSVersion
Value : 5.1.17763.4974
Name  : PSVersion

Key   : PSEdition
Value : Desktop
Name  : PSEdition

Key   : PSCompatibleVersions
Value : {1.0, 2.0, 3.0, 4.0...}
Name  : PSCompatibleVersions

Key   : BuildVersion
Value : 10.0.17763.4974
Name  : BuildVersion

Key   : CLRVersion
Value : 4.0.30319.42000
Name  : CLRVersion

Key   : WSManStackVersion
Value : 3.0
Name  : WSManStackVersion

Key   : PSRemotingProtocolVersion
Value : 2.3
Name  : PSRemotingProtocolVersion

Key   : SerializationVersion
Value : 1.1.0.1
Name  : SerializationVersion
```

**Comment by skghq** [comment]
I have upgraded to 1.23.1122.1. This is still an issue.

**Comment by andikrueger** [comment]
Please make sure to add RoleEligibilitySchedule.ReadWrite.Directory to your application. This permission is listed as requirement within the resource's setting.json file.

**Comment by mlhickey** [comment]
I noticed that RoleEligibilitySchedule.ReadWrite.Directory is listed for both read and update - shouldn't read it be:
`
            "application": {
                "read": [
                    {
                        "name": "RoleEligibilitySchedule.Read.Directory"
                    }
                ],
                "update": [
                    {
                        "name": "RoleEligibilitySchedule.ReadWrite.Directory"
                    }
                ]
            }
`

**Comment by andikrueger** [comment]
This got fixed within the latest release by #4326

**Comment by skghq** [comment]
I have upgraded to 1.24.228.1 and it's still an issue @andikrueger. My "main" export is still running. Once it completes I will go back and manually add the permission.

`[2024/03/01 04:42:04]
{InvalidOperation}
System.Exception: [UnknownError] : {"errorCode":"PermissionScopeNotGranted","message":"Authorization failed due to missing permission scope RoleEligibilitySchedule.ReadWrite.Directory,RoleManagement.ReadWrite.Directory.","instanceAnnotations":[]}
"Error during Export:"
at Get-MgBetaRoleManagementDirectoryRoleEligibilityScheduleRequest<Process>, C:\Program Files\WindowsPowerShell\Modules\Microsoft.Graph.Beta.Identity.Governance\2.14.1\exports\ProxyCmdletDefinitions.ps1: line 122870
at Export-TargetResource, C:\Program Files\WindowsPowerShell\Modules\Microsoft365DSC\1.24.228.1\DSCResources\MSFT_AADRoleEligibilityScheduleRequest\MSFT_AADRoleEligibilityScheduleRequest.psm1: line 800
at Start-M365DSCConfigurationExtract, C:\Program Files\WindowsPowerShell\Modules\Microsoft365DSC\1.24.228.1\Modules\M365DSCReverse.psm1: line 649
at Export-M365DSCConfiguration, C:\Program Files\WindowsPowerShell\Modules\Microsoft365DSC\1.24.228.1\Modules\M365DSCUtil.psm1: line 1357
at <ScriptBlock>, C:\Scripts\username\From-server\Office365_Backup_Debug.ps1: line 778`
